src/runtime/image.ts

```diff
@@ -182,16 +184,47 @@ function getSizes (ctx: ImageCTX, input: string, opts: ImageSizesOptions) {
     })
   }
 
-  variants.sort((v1, v2) => v1.screenMaxWidth - v2.screenMaxWidth)
+  sizeVarients.sort((v1, v2) => v1.screenMaxWidth - v2.screenMaxWidth)
 
-  const defaultVar = variants[variants.length - 1]
+  const defaultVar = sizeVarients[sizeVarients.length - 1]
   if (defaultVar) {
     defaultVar.media = ''
   }
 
+  // Only use srcset if it is a string
+  if (typeof opts.srcset === 'string') {
+    for (const entry of opts.srcset.split(/[ ,]+/).filter(e => e)) {
+      if (/^[x]\d+/.test(entry)) {
```

Comment:
The regex matches on x9999999 - maybe make it so that it rejects anything more than a single digit?

Reply:
Sorry, took a while to get around to this. The regex now checks to make sure that only a single digit is used in the descriptor

Reply:
Thanks for making the changes!